pyresample/bilinear/__init__.py

```diff
@@ -210,97 +203,448 @@ def get_bil_info(source_geo_def, target_area_def, radius=50e3, neighbours=32,
+        """Resample using pre-computed resampling LUTs."""
+        del output_shape
+        fill_value = _check_fill_value(fill_value, data.dtype)
+
+        res = _resample(
+            self._slice_data(data, fill_value),
+            (self.bilinear_s, self.bilinear_t)
+        )
+
+        return self._finalize_output_data(data, res, fill_value)
+
+    def _slice_data(self, data, fill_value):
+        return get_slicer(data)(data, self.slices_x, self.slices_y, self.mask_slices, fill_value)
+
+    def _finalize_output_data(self, data, res, fill_value):
+        return res
+
+    def save_bil_info(self, filename):
+        """Save bilinear resampling look-up tables."""
+        raise NotImplementedError
+
+    def load_bil_info(self, filename):
+        """Load bilinear resampling look-up tables and initialize the resampler."""
+        raise NotImplementedError
+
+
+def _check_fill_value(fill_value, dtype):
+    """Check that fill value is usable for the data."""
+    if fill_value is None:
+        if np.issubdtype(dtype, np.integer):
+            fill_value = 0
+        else:
+            fill_value = np.nan
+    elif np.issubdtype(dtype, np.integer):
+        if np.isnan(fill_value):
+            fill_value = 0
+        elif np.issubdtype(type(fill_value), np.floating):
+            fill_value = int(fill_value)
+
+    return fill_value
+
+
+def _get_output_xy(target_geo_def):
+    out_x, out_y = target_geo_def.get_proj_coords()
+    return np.ravel(out_x),  np.ravel(out_y)
+
+
+def _get_input_xy(source_geo_def, proj, valid_input_index, index_array):
+    """Get x/y coordinates for the input area and reduce the data."""
+    return proj(
+        *array_slice_for_multiple_arrays(
+            index_array,
+            array_slice_for_multiple_arrays(
+                valid_input_index,
+                mask_coordinates(*_get_raveled_lonlats(source_geo_def))
+            )
+        )
+    )
+
+
+def array_slice_for_multiple_arrays(idxs, data):
+    """Slices multiple arrays using the same indices."""
+    return [d[idxs] for d in data]
+
+
+def mask_coordinates(lons, lats):
+    """Mask invalid coordinate values."""
+    idxs = (find_indices_outside_min_and_max(lons, -180., 180.) |
+            find_indices_outside_min_and_max(lats, -90., 90.))
+    return _np_where_for_multiple_arrays(idxs, (np.nan, np.nan), (lons, lats))
+
+
+def find_indices_outside_min_and_max(data, min_val, max_val):
+    """Return array indices outside the given minimum and maximum values."""
+    return (data < min_val) | (data > max_val)
+
+
+def _get_raveled_lonlats(geo_def):
+    lons, lats = geo_def.get_lonlats()
+    if lons.size == 0 or lats.size == 0:
+        raise ValueError('Cannot resample empty data set')
+    elif lons.size != lats.size or lons.shape != lats.shape:
+        raise ValueError('Mismatch between lons and lats')
+
+    return np.ravel(lons), np.ravel(lats)
+
+
+def _get_bounding_corners(in_x, in_y, out_x, out_y, neighbours, index_array):
```

Comment:
Should this be renamed `_get_four_closest_corners`?

Reply:
Yes, thanks.

Reply:
Done.